tests/framework/data/ActiveDataProviderTest.php

```diff
@@ -177,4 +178,42 @@ public function testPaginationBeforeModels()
         $provider->refresh();
         $this->assertEquals(2, count($provider->getModels()));
     }
+
+    public function testZeroModels()
+    {
+        $query = new UnqueryableQueryMock;
+        $provider = new ActiveDataProvider([
+            'db' => $this->getConnection(),
+            'query' => $query->from('order')->where('0=1'),
+        ]);
+        $pagination = $provider->getPagination();
+        $this->assertEquals(0, $pagination->getPageCount());
+
+        try {
+            $this->assertCount(0, $provider->getModels());
+        } catch (InvalidCallException $exception) {
+            $this->fail('An excessive models query was executed.');
+        }
+
+        $this->assertEquals(0, $pagination->getPageCount());
+    }
+}
+
+class UnqueryableQueryMock extends Query
```

Comment:
I dont have a problem with keeping it in the current file as it is unlikely to be reused in other tests. Additionally it should be replaced by an anonymous class in PHP7 in the (hopefully near) future